File: 2155-find-missing-observations/find-missing-observations.py
class Solution:
    def missingRolls(self, rolls: List[int], mean: int, n: int) -> List[int]:
        total_elements = len(rolls) + n
        sum_of_all_elements = total_elements * mean
        sum_of_second_list = sum_of_all_elements - sum(rolls)
        if sum_of_second_list > 6 * n or sum_of_second_list < n:
            return []

        quotient, remainder = divmod(sum_of_second_list, n)
        return [quotient + (1 if i < remainder else 0) for i in range(n)]
            

File: 2155-find-missing-observations/find-missing-observations.java
class Solution {
    public int[] missingRolls(int[] rolls, int mean, int n) {
        int[] result = new int[n];
        int sum = 0;
        for (int i : rolls) 
            sum += i;

        int missingSum = mean * (n + rolls.length) - sum;
        if (missingSum > 6 * n || missingSum < n) 
            return new int[] {};

        int expectedAvg = missingSum / n, remainder = missingSum % n;

        for (int i = 0; i < remainder; i++) 
            result[i] = expectedAvg + 1;
        for (int i = remainder; i < n; i++) 
            result[i] = expectedAvg;
        
        return result;

    }
}